File: DataAlgo_forPy/Stacks_Queues/queue.rb
class Queue
  def initialize
    @items = []
  end

  def enqueue(item)
    @items.unshift(item)
  end

  def dequeue
    @items.pop()
  end

  def is_empty?
    @items.empty?
  end

  def size
    @items.length
  end

end

myQueue = Queue.new

p myQueue.is_empty?

myQueue.enqueue(1)
myQueue.enqueue(2)
myQueue.enqueue("three")

p myQueue.size

p myQueue.dequeue
myQueue.dequeue
myQueue.dequeue

p myQueue.is_empty?

File: DataAlgo_forPy/Stacks_Queues/queue.js
class Queue {
  constructor() {
    this.items = []
  }

  isEmpty() {
    return this.items.length === 0
  }

  enqueue(item) {
    this.items.unshift(item)
  }

  dequeue() {
    return this.items.pop()
  }

  size() {
    return this.items.length
  }
}

const myQueue = new Queue()

console.log(myQueue.isEmpty())

myQueue.enqueue(1)
myQueue.enqueue(2)

console.log(myQueue.dequeue())
